x/bep3/types/params_test.go

```diff
@@ -46,189 +41,327 @@ func (suite *ParamsTestSuite) TestParamValidation() {
 		{
-			name: "minimum block lock greater than maximum block lock",
+			name: "invalid denom - bad format",
 			args: args{
-				bnbDeputyAddress:  suite.addr,
-				bnbDeputyFixedFee: types.DefaultBnbDeputyFixedFee,
-				minAmount:         types.DefaultMinAmount,
-				maxAmount:         types.DefaultMaxAmount,
-				minBlockLock:      500,
-				maxBlockLock:      400,
-				supportedAssets:   types.DefaultSupportedAssets,
+				assetParams: types.AssetParams{types.NewAssetParam(
+					"BNB", 714, suite.supply[0], true,
+					suite.addr, sdk.NewInt(1000), sdk.NewInt(100000000), sdk.NewInt(100000000000),
+					types.DefaultMinBlockLock, types.DefaultMaxBlockLock)},
 			},
 			expectPass:  false,
-			expectedErr: "minimum block lock cannot be > maximum block lock",
+			expectedErr: "denom invalid",
 		},
 		{
-			name: "empty asset denom",
+			name: "min block lock equal max block lock",
 			args: args{
-				bnbDeputyAddress:  suite.addr,
-				bnbDeputyFixedFee: types.DefaultBnbDeputyFixedFee,
-				minAmount:         types.DefaultMinAmount,
-				maxAmount:         types.DefaultMaxAmount,
-				minBlockLock:      types.DefaultMinBlockLock,
-				maxBlockLock:      types.DefaultMaxBlockLock,
-				supportedAssets: types.AssetParams{
-					types.AssetParam{
-						Denom:  "",
-						CoinID: 714,
-						Limit:  sdk.NewInt(100000000000),
-						Active: true,
-					},
-				},
+				assetParams: types.AssetParams{types.NewAssetParam(
+					"bnb", 714, suite.supply[0], true,
+					suite.addr, sdk.NewInt(1000), sdk.NewInt(100000000), sdk.NewInt(100000000000),
+					243, 243)},
+			},
+			expectPass:  true,
+			expectedErr: "",
+		},
+		{
+			name: "min block lock greater max block lock",
+			args: args{
+				assetParams: types.AssetParams{types.NewAssetParam(
+					"bnb", 714, suite.supply[0], true,
+					suite.addr, sdk.NewInt(1000), sdk.NewInt(100000000), sdk.NewInt(100000000000),
+					244, 243)},
 			},
 			expectPass:  false,
-			expectedErr: "asset denom cannot be empty",
+			expectedErr: "minimum block lock > maximum block lock",
 		},
 		{
-			name: "negative asset coin ID",
+			name: "min swap not positive",
 			args: args{
-				bnbDeputyAddress:  suite.addr,
-				bnbDeputyFixedFee: types.DefaultBnbDeputyFixedFee,
-				minAmount:         types.DefaultMinAmount,
-				maxAmount:         types.DefaultMaxAmount,
-				minBlockLock:      types.DefaultMinBlockLock,
-				maxBlockLock:      types.DefaultMaxBlockLock,
-				supportedAssets: types.AssetParams{
-					types.AssetParam{
-						Denom:  "bnb",
-						CoinID: -1,
-						Limit:  sdk.NewInt(100000000000),
-						Active: true,
-					},
-				},
+				assetParams: types.AssetParams{types.NewAssetParam(
+					"bnb", 714, suite.supply[0], true,
+					suite.addr, sdk.NewInt(1000), sdk.NewInt(0), sdk.NewInt(10000000000),
+					types.DefaultMinBlockLock, types.DefaultMaxBlockLock)},
 			},
 			expectPass:  false,
-			expectedErr: "must be a non negative integer",
+			expectedErr: "must have a positive minimum swap",
```

Comment:
Can we update this test to `MinSwapAmount: sdk.ZeroInt()` once the validation function has been updated?

Reply:
Marking as resolved due to https://github.com/Kava-Labs/kava/pull/616#discussion_r465308543